### PR TITLE
Ellen Nitchals

### PR DESCRIPTION
**Pull request by enitchals**

### CODE REVIEW

**Comment by ckopecky** [comment]
Hi, Ellen! I'll just reiterate what Ryan said in your closed PR since there's not really much difference between the code from the closed PR and the open PR. 

It's interesting reviewing code from a person you once looked to for guidance in helping you write your own code. :) I learned a lot from you and a lot of my ability to write good, clean code has come from your ability to write good, clean code! 

Well done!

**Comment by mixelpixel** [comment]
@ckopecky test

File: api/controllers/budgetControllers.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const mongoose = require('mongoose');
+
+const Budget = require('../models/budget');
+
+const createBudget = (req, res) => {
+    const { title, amount } = req.body;
+    const newBudget = new Budget({ title, amount });
+    newBudget.save((err, savedBudget) => {
+        if(err) {
+            res.status(500).json(err);
```
Comment (mixelpixel) [inline api/controllers/budgetControllers.js]: @enitchals comment directly on the code

File: Answers.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+1. Describe the following: DataBase, Collection , Document
+A database is the top-level (app-level) organization of data on a Mongo server.
+A collection is a subsection of a database, and it helps to organize the documents within the database.
+A document is a collection of data, organized as a JavaScript object.
+
```
Comment (ckopecky) [inline Answers.md:5]: Good overview of the difference between DB, Coll, Document!

File: Answers.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+2. Describe how one can achieve the pattern of relationships in MongoDB. What needs to take place at the schema level? How do we 'fill' in the appropriate relational data using mongoose?
+In the Schema, we specify that a field type using mongoose.Schema.Types.ObjectId and then pass a reference (ref) to the Schema.
+We can use Mongoose's populate function to auto-populate ID references, including the referenced document within the larger returned document.
+
+3. Describe what MVC Archtecture is and how we have used it this week with Node/Express/Mongoose.
+MVC = Model View Controller, a pattern for building UIs.
+In the Model, we use Mongo(ose) to build a schema that serves as a data model -- a blueprint/template for what we want our data to look like.
+Controllers interpret a request, perform operations on the database, and send a response.
+Our View is the UI itself -- the parts of our stack that generate and deliver content based on the data delivered in the Controller responses.
```
Comment (ckopecky) [inline Answers.md:14]: good response.

File: api/routes/routes.js [hunk 4]
```diff
@@ -1,3 +1,14 @@
...
-  // Todo: Fill in your routes here
+  app.route('/budget')
+    .post(budgetControllers.createBudget);
+  app.route('/category')
+    .get(categoryControllers.listCategories)
+    .post(categoryControllers.createCategory);
+  app.route('/expense')
+    .get(expenseControllers.listExpenses)
+    .post(expenseControllers.createExpense);
 };
```
Comment (ckopecky) [inline api/routes/routes.js:14]: Overall, really clean code! I'm happy to have been able to take a look! Just double check those commit messages and make them more meaningful so your future self will thank you :)

**Review by ckopecky — APPROVED** [review]

**Pull request closed without merging**